src/test/java/org/apache/commons/imaging/formats/jpeg/exif/ExifRewriterRoundtripTest.java

```diff
@@ -56,124 +62,160 @@ public static Stream<File> data() throws Exception {
+
+            if (!fieldCountMatches) {
+
+                /*
+                 * Note that offset fields are not written again if a re-order makes
+                 * them unnecessary. The TiffWriter does not write in the same order,
+                 * but tries to optimize it.
+                 */
+
+                final List<Integer> sourceTags = new ArrayList<>();
+                final List<Integer> duplicatedTags = new ArrayList<>();
+
+                for (TiffOutputField field : sourceFields)
+                    sourceTags.add(field.tag);
+
+                for (TiffOutputField field : duplicateFields)
+                    duplicatedTags.add(field.tag);
+
+                final List<Integer> missingTags = new ArrayList<>(sourceTags);
+                missingTags.removeAll(duplicatedTags);
+                missingTags.remove((Integer) ExifTagConstants.EXIF_TAG_EXIF_OFFSET.tag);
+                missingTags.remove((Integer) ExifTagConstants.EXIF_TAG_GPSINFO.tag);
+                missingTags.remove((Integer) ExifTagConstants.EXIF_TAG_INTEROP_OFFSET.tag);
+                missingTags.remove((Integer) TiffTagConstants.TIFF_TAG_JPEG_INTERCHANGE_FORMAT.tag);
+
+                assertTrue(missingTags.isEmpty(), "Missing tags: " + missingTags);
+            }
+
+            for (TiffOutputField sourceField : sourceFields) {
+
+                final boolean isOffsetField =
+                        sourceField.tag == ExifTagConstants.EXIF_TAG_EXIF_OFFSET.tag ||
+                                sourceField.tag == ExifTagConstants.EXIF_TAG_GPSINFO.tag ||
+                                sourceField.tag == ExifTagConstants.EXIF_TAG_INTEROP_OFFSET.tag ||
+                                sourceField.tag == TiffTagConstants.TIFF_TAG_JPEG_INTERCHANGE_FORMAT.tag;
+
+                /*
+                 * Ignore offset fields. They may not be needed after rewrite
+                 * and their value changes anyway.
+                 */
+                if (isOffsetField)
+                    continue;
+
+                TiffOutputField duplicateField = duplicateDirectory.findField(sourceField.tag);
+
+                assertNotNull(duplicateField, "Field is missing: " + sourceField.tagInfo);
+
+                assertEquals(sourceField.tag, duplicateField.tag, "TiffOutputField tag mismatch.");
+                assertEquals(sourceField.abstractFieldType, duplicateField.abstractFieldType, "TiffOutputField fieldType mismatch.");
+                assertEquals(sourceField.count, duplicateField.count, "TiffOutputField count mismatch.");
+
+                boolean sameValue = TiffOutputFieldTestUtil.bytesEqual(sourceField, duplicateField);
+
+                assertTrue(sameValue, "Bytes are different for field: " + sourceField);
             }
         }
     }
 
     private void copyToDuplicateFile(final File sourceFile, final TiffOutputSet duplicateTiffOutputSet) throws IOException, ImagingException, ImagingException {
+
```

Comment:
Don't change whitespace, please. It clutters up the PR with noise and makes it harder to review.

Reply:
It's a really small PR and this empty line makes the code easier to read.